xds/src/main/java/io/grpc/xds/CdsLoadBalancer2.java

```diff
@@ -181,9 +185,17 @@ private void handleClusterDiscovered() {
         return;
       }
 
+      // The LB policy config is provided in service_config.proto/JSON format. It is unwrapped
+      // to determine the name of the policy in the load balancer registry.
+      LbConfig unwrappedLbConfig = ServiceConfigUtil.unwrapLoadBalancingConfig(
+          root.result.lbPolicyConfig());
+      LoadBalancerProvider lbProvider = lbRegistry.getProvider(unwrappedLbConfig.getPolicyName());
+      Object lbConfig = lbProvider.parseLoadBalancingPolicyConfig(
```

Comment:
I do think we should double-check the parse was successful, as otherwise we're going to get a surprise NPE much later in the tree. I'm okay if we just throw (IllegalArgumentException?); that will completely break the channel, but seeing the error early here in the unexpected case it occurs is the main goal.

Reply:
I added two checks. One makes sure that the provider parses the config ok. If it returns an error it throws StatusRuntimeException based on the returned error status. The second check is for the existence of a provider, I use the same exception type for consistency. Also added tests for these two cases.